### PR TITLE
fixing Flash\Session::has() returning always true

### LINKED ISSUE #15204: Phalcon\Flash\Session::has return true when messages array is empty
Hi,

Using `Phalcon\Flash\Session::has()` return true whereas the messages array is empty.

https://github.com/phalcon/cphalcon/blob/a8035814aa9a8a2438f92ebf5128964e7c48fd3f/phalcon/Flash/Session.zep#L43

```
/**
     * Checks whether there are messages
     */
    public function has(type = null) -> bool
    {
        var messages;

        let messages = this->getSessionMessages(false);

        if typeof type != "string" {
            return true;
        }

        return isset messages[type];
    }
```

Sorry I don't understand why `type != 'string` return true,
but maybe it's possible to add something like :
```
if empty messages {
    return false;
}
```

Regards,

### PR DESCRIPTION
**Pull request by Izopi4a**
Hello!

* Type: bug fix 
* Link to issue: https://github.com/phalcon/cphalcon/issues/15204

**In raising this pull request, I confirm the following:**

- [x] I have read and understood the [Contributing Guidelines](https://github.com/phalcon/cphalcon/blob/master/CONTRIBUTING.md)
- [x] I have checked that another pull request for this purpose does not exist
- [x] I wrote some tests for this PR
- [x] I have updated the relevant CHANGELOG
- [ ] I have created a PR for the [documentation](https://github.com/phalcon/docs) about this change

Small description of change:

I am basically reverting the changes on Flash\Session::has() from this PR https://github.com/phalcon/cphalcon/pull/13994/files

I don't agree with the way funtion is changed, feels like just a bug

Thanks

### CODE REVIEW

File: phalcon/Flash/Session.zep [hunk 1]
```diff
@@ -39,18 +39,22 @@ class Session extends AbstractFlash
 
     /**
      * Checks whether there are messages
+     * 
+     * @param string|null $type
+     *
+     * @return bool
      */
-    public function has(type = null) -> bool
+    public function has(var type = null) -> bool
```
Comment (Jeckerson) [inline phalcon/Flash/Session.zep:47]:
```suggestion
    public function has(string type = null) -> bool
```

**Review by niden — DISMISSED** [review]

**Comment by Izopi4a** [comment]
i will check to see what is wrong, probably i broke something , changed the MR name to wip

**Comment by Izopi4a** [comment]
ok, so if the declarion in zephir is as followed:

```
public function test(string test = null) {
    var_dump(test);
}

test(); // will output string ""
```

**Comment by Jeckerson** [comment]
@Izopi4a Yep, just simulated in Zephir. For now we can specify `var` and check type if is string.

https://github.com/zephir-lang/zephir/issues/2180

**Review by Jeckerson — APPROVED** [review]

**Review by niden — APPROVED** [review]

**Comment by niden** [comment]
Fixes https://github.com/phalcon/cphalcon/pull/15353#pullrequestreview-625438556

@Izopi4a Thank you!

**Pull request merged**